x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/140_metadata.yml

```
@@ -170,4 +170,43 @@ setup:
       catch: /cannot sort on _source/
       esql.query:
         body:
-          query: 'FROM test metadata _source | sort _source'
+          query: 'FROM test metadata _source | SORT _source'
+
+---
+"sort returning _source is allowed":
+  - requires:
+      test_runner_features: [capabilities]
+      capabilities:
+        - method: POST
+          path: /_query
+          parameters: []
+          capabilities: [sorting_on_source_and_counters_forbidden]
+      reason: "Sorting on _source shouldn't have been possible"
+  - do:
+      esql.query:
+        body:
+          query: 'FROM test metadata _source | SORT case ASC | KEEP case, _source | LIMIT 5'
```

nitty nit, for a next PR that might touch this (here and above)
```suggestion
          query: 'FROM test METADATA _source | SORT case ASC | KEEP case, _source | LIMIT 5'
```